protocol/x/vault/types/events.go

```diff
@@ -0,0 +1,51 @@
+package types
+
+import (
+	fmt "fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	// satypes "github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types"
+)
+
+const (
+	EventTypeDepositToMegavault = "deposit_to_megavault"
+	AttributeKeyDepositor       = "depositor"
+	AttributeKeyQuoteQuantums   = "quote_quantums"
+	AttributeKeyMintedShares    = "minted_shares"
+
+	EventTypeWithdrawFromMegavault    = "withdraw_from_megavault"
+	AttributeKeyWithdrawer            = "withdrawer"
+	AttributeKeySharesToWithdraw      = "shares_to_withdraw"
+	AttributeKeyTotalShares           = "total_shares"
+	AttributeKeyRedeemedQuoteQuantums = "redeemed_quote_quantums"
+)
+
+// NewDepositToMegavaultEvent constructs a new deposit_to_megavault sdk.Event.
+func NewDepositToMegavaultEvent(
+	depositorAddress string,
+	quoteQuantums uint64,
+	mintedShares uint64,
+) sdk.Event {
+	return sdk.NewEvent(
+		EventTypeDepositToMegavault,
+		sdk.NewAttribute(AttributeKeyDepositor, depositorAddress),
+		sdk.NewAttribute(AttributeKeyQuoteQuantums, fmt.Sprintf("%d", quoteQuantums)),
+		sdk.NewAttribute(AttributeKeyMintedShares, fmt.Sprintf("%d", mintedShares)),
+	)
+}
+
+// NewWithdrawFromMegavaultEvent constructs a new withdraw_from_megavault sdk.Event.
+func NewWithdrawFromMegavaultEvent(
```

Comment:
non-blocking, could add in total equity of the megavault in the event, so that on it's own we can calculate the aggregate slippage of the withdraw.